ttnn/cpp/ttnn/operations/data_movement/common/common.hpp

```diff
@@ -156,6 +156,27 @@ ttnn::Tensor pad_to_tile_vol(
     const bool use_multicore,
     const std::optional<MemoryConfig>& memory_config);
 
+enum class ShardStrategy { BLOCK, HEIGHT, WIDTH };
+
+// Helper function for creating a sharded memory configuration for a tensor
+// based on its logical shape, a shard strategy and orientation, and a core
+// grid. Optionally, you may pass a preferred shard shape to use. If not
+// provided, the shard shape will be inferred from the tensor shape and the
+// shard strategy.
+ttnn::MemoryConfig create_sharded_memory_config(
+    const ttnn::SimpleShape& logical_shape,
+    const tt::tt_metal::CoreRangeSet& core_grid,
+    const ShardStrategy& strategy,
+    const tt::tt_metal::ShardOrientation& orientation,
+    std::optional<std::array<uint32_t, 2>> shard_shape = std::nullopt,
+    const tt::tt_metal::Layout& layout = tt::tt_metal::Layout::ROW_MAJOR,
+    bool halo = false);
```

Comment:
halo was recently removed from memory_config, so I think this must be removed

Reply:
I still see halo in the ShardSpec config.

Reply:
Let's revisit this once halo is removed from ShardSpec.